atuin/src/command/client/history.rs

```diff
@@ -126,8 +136,16 @@ pub fn print_list(h: &[History], list_mode: ListMode, format: Option<&str>, reve
         Box::new(h.iter()) as Box<dyn Iterator<Item = &History>>
     };
 
+    let entry_terminator = if print0 { "\0" } else { "\n" };
```

Comment:
No newline with nulls seems weird. Is that actually how the other tools work?

Reply:
Yes; the point of this is that if the string might contain newlines -- say, if the command does -- then this allows linewise processors to know where the boundaries of the command are.

The common pattern here is `find -print0 | xargs -0` which allows for filenames with all sorts of bonkers delimiter-like characters.

Reply:
I _do_ have one rust-ish question on this, though; I'm used to IO that flushes on newlines, which is why I am only calling `flush` when the delimiter is null. However, I'm not entirely sure if the output flushing for `write!` will behave that way, or if I'd need to flush when calling `write!(...something with a newline)` manually.

Reply:
You are correct. write for std::io::stdout will flush on  newlines automatically. What you are doing here is equivalent

Reply:
Perfect. Well, let me know if you want any other details around this change; I'd love to get multiline history search in my shell again, and this is a key step)

Reply:
> I'd love to get multiline history search in my shell again

Mind if I ask where we're lacking in that regard?

Reply:
Honestly... I don't like the built-in search. I find the search tools fzf offers more enjoyable from my side, so finding a way to get atuin and fzf to play nice is my current goal. 

So, I've been working on history search functions that combine the output of `atuin`, a bit of null-terminated `gawk` scripting, and `fzf` to do the search.